### DISCUSSION TITLE
Cascading deletes

### DISCUSSION THREAD

**Discussion by divan** [geldata/gel#4045] [General]
Sorry for lame question, this should be clear from documentation, but I'm having troubles defining proper link deletion policies.

Let's say we have two entities - User and Post. User can have multiple posts.

I want to ensure that all user's posts are automatically deleted whenever user is deleted.

Schema is:
```
module default {
    type User {
        required property name -> str;
        multi link posts -> Post {
            constraint exclusive;
        }
    }

    type Post {
        required property text -> str;
    }
}
```

I understand that if link is defined in opposite direction (i.e. `Post` as property `author` with `on targe delete delete source`  polity), then it should work. But what to do in this case?

I'm guessing some solutions like:

 - manually delete posts first, and then delete user (safe, but almost unusable for the large schema)
 - maybe define 'author' property on `Post` with backlinks and define target deletion policy there? (feels hackish and not sure what's happening under the hood)

Would be grateful for advice on how to idiomatically implement such a deletion policy.

**Comment by elprans** [comment]
2.0 has support for two more policy types: `on source delete delete target` and `on source delete delete target if orphan`, which cover this use case.

**Comment by divan** [comment]
@elprans thanks! Good to know it's not me missing something obivious. Eagerly awaiting July 28th then!
